scripts/bug_retriever.py

```diff
@@ -17,26 +17,39 @@ class Retriever(object):
     def retrieve_bugs(self, limit: int = None) -> None:
         bugzilla.set_token(get_secret("BUGZILLA_TOKEN"))
 
+        last_modified = None
         db.download(bugzilla.BUGS_DB)
 
         # Get IDs of bugs changed since last run.
-        last_modified = db.last_modified(bugzilla.BUGS_DB)
-        logger.info(
-            f"Retrieving IDs of bugs modified since the last run on {last_modified}"
-        )
-        changed_ids = set(
-            bugzilla.get_ids(
-                {"f1": "delta_ts", "o1": "greaterthaneq", "v1": last_modified.date()}
+        try:
+            last_modified = db.last_modified(bugzilla.BUGS_DB)
+        except Exception as e:
+            if str(e) == "Last-Modified is not available":
+                pass
+
+        if last_modified:
+            logger.info(
+                f"Retrieving IDs of bugs modified since the last run on {last_modified}"
             )
-        )
+            changed_ids = set(
+                bugzilla.get_ids(
+                    {
+                        "f1": "delta_ts",
+                        "o1": "greaterthaneq",
+                        "v1": last_modified.date(),
+                    }
+                )
+            )
+
         logger.info(f"Retrieved {len(changed_ids)} IDs.")
 
         all_components = bugzilla.get_product_component_count(9999)
 
         deleted_component_ids = set(
             bug["id"]
             for bug in bugzilla.get_bugs()
-            if "{}::{}".format(bug["product"], bug["component"]) not in all_components
+            if "{}::{}".format(bug["product"], bug["component"])
+            not in all_components
```

Comment:
Please change this back to the old formatting

Reply:
Actually pre-commit changed it before.
black check in pre-commit is not working properly on my system.
It some times working. But not all the times.

Ok, I will change it manually.

Reply:
Done